packages/vx-axis/src/axis/AxisRenderer.tsx

```diff
@@ -30,51 +29,36 @@ export default function AxisRenderer<Scale extends AxisScale>({
   orientation,
   scale,
   stroke = '#222',
-  strokeWidth = 1,
   strokeDasharray,
+  strokeWidth = 1,
   tickClassName,
   tickComponent,
   tickLabelProps = (/** tickValue, index */) => defaultTextProps,
   tickLength,
   tickStroke = '#222',
   tickTransform,
   ticks,
+  ticksComponent = TicksRenderer,
 }: AxisRendererProps<Scale>) {
-  let tickLabelFontSize = 10; // track the max tick label size to compute label offset
-
+  // compute the max tick label size to compute label offset
+  const allTickLabelProps = ticks.map(({ value, index }) => tickLabelProps(value, index));
```

Comment:
this is the jank logic I alluded to above. The label and ticks were previously coupled with this, so I updated the logic to pass an array of `tickLabelProps` to `ticksComponent`. 

Might be worth a breaking change to remove this?